feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/captive.c

```diff
@@ -91,65 +98,96 @@ const struct uci_blob_param_list dnsm_param = {
 	.params = dnsm_policy,
 };
 
-void vif_state_dhcp_allowlist_get(struct schema_Wifi_VIF_State *vstate,struct uci_section *s)
+/*void removeSpaces(char *str)
 {
+	int count = 0;
+	for (int i = 0; str[i]; i++)
+		if (str[i] != ' ')
+			str[count++] = str[i];
+	str[count] = '\0';
+}*/
+
+void vif_state_dhcp_allowlist_get(struct schema_Wifi_VIF_State *vstate, char *section)
+{
+	char fqdn[32];
+	char read_ifname[8];
+	char set[8];
 	struct blob_attr *td[__DNS_ATTR_MAX] = { };
-	struct uci_element *e = NULL;
-
-	uci_load(dns, "dhcp", &dhcp);
-	uci_foreach_element(&dhcp->sections, e) {
-		struct uci_section *ip_section = uci_to_section(e);
-		if (!strcmp(s->e.name, ip_section->e.name)){
-
-			blob_buf_init(&dnsmas, 0);
-			uci_to_blob(&dnsmas, ip_section, &dnsm_param);
-			blobmsg_parse(dnsm_policy, __DNS_ATTR_MAX, td, blob_data(dnsmas.head), blob_len(dnsmas.head));
-
-			if (td[DNS_ATTR_IPSET]) {
-				struct blob_attr *cur = NULL;
-				int rem = 0;
-				vstate->captive_allowlist_len = 0;
-				blobmsg_for_each_attr(cur, td[DNS_ATTR_IPSET], rem) {
-					if (blobmsg_type(cur) != BLOBMSG_TYPE_STRING)
-						continue;
-					strcpy(vstate->captive_allowlist[vstate->captive_allowlist_len], blobmsg_get_string(cur));
-
-					vstate->captive_allowlist_len++;
-				}
+	uci_load(dns, "dhcp", &dhc);
+	ip_section = uci_lookup_section(dns, dhc, section);
+	if(!ip_section) {
+		LOGN("Section Not Found");
+		return;
+	}
+	blob_buf_init(&dnsmas, 0);
+	uci_to_blob(&dnsmas, ip_section, &dnsm_param);
+	blobmsg_parse(dnsm_policy, __DNS_ATTR_MAX, td, blob_data(dnsmas.head), blob_len(dnsmas.head));
+
+	if (td[DNS_ATTR_IPSET]) {
+
+		int rem = 0;
+		vstate->captive_allowlist_len = 0;
+		blobmsg_for_each_attr(cur, td[DNS_ATTR_IPSET], rem) {
+			if (blobmsg_type(cur) != BLOBMSG_TYPE_STRING)
+				continue;
+			sscanf(blobmsg_get_string(cur), "/%[^/]/%[^_]_%s", fqdn, set, read_ifname);
+			if(!strcmp(vstate->if_name, read_ifname))
+			{
+				strcpy(vstate->captive_allowlist[vstate->captive_allowlist_len], fqdn);
+				vstate->captive_allowlist_len++;
 			}
 		}
 	}
-uci_unload(dns, dhcp);
-return;
+	uci_unload(dns, dhc);
+	return;
 }
-int ipset_create(char *ifname)
+
+/*bool isIpAddresscorrect(char *ipAddress)
```

Comment:
more unused code that can most likely be dropped

Reply:
sure, In the future I'm going to use that piece of code ,so I commented for now. but will delete that for now